packages/firestore/test/integration/api/get_options.test.ts

```diff
@@ -0,0 +1,612 @@
+/**
+ * Copyright 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { expect } from 'chai';
+import { apiDescribe, withTestDoc, withTestCollection } from '../util/helpers';
+
+apiDescribe('GetOptions', persistence => {
```

Comment:
Thanks for all the tests. I love good test coverage! One thing I don't love though is boilerplate and lots of code to port though...  So you might consider if:
1) Any of these tests are redundant
2) If you could simplify them (and reduce boilerplate) with helper method(s).
3) If you could get equivalent coverage but with less boilerplate by testing at a lower-level. [this probably doesn't apply here, but in general we try to do our more exhaustive testing as "unit" tests which run much more quickly and then only sanity-checking at the API layer]